### ISSUE TITLE
python ModuleNotFoundError: No module named 'mmlspark.lightgbm._LightGBMClassifier'

### ISSUE THREAD

**Issue by rusonding** [Azure/mmlspark#935]
**Describe the bug**
ModuleNotFoundError: No module named 'mmlspark.lightgbm._LightGBMClassifier'

**To Reproduce**
 spark2-submit --master local --jars file:///root/.ivy2/jars/*  --conf spark.pyspark.python=/usr/lib/anaconda2/envs/mmlspark/bin/python --num-executors 5 --executor-memory 2G test_mmlspark.py

**Expected behavior**

**Info (please complete the following information):**
 - MMLSpark Version: [ 0.0.11111111
 - Spark Version [2.4.5]
 - Spark Platform [cdh]

- certifi         2016.2.28
- future          0.18.2
- mmlspark        0.0.11111111
- numpy           1.19.2
- pip             20.2.3
- py4j            0.10.7
- PyHive          0.6.1
- pyspark         2.4.5
- python-dateutil 2.8.1
- setuptools      36.4.0
- six             1.15.0
- wheel           0.29.0

** Stacktrace**
```
Traceback (most recent call last):
  File "/root/test/test_mmlspark.py", line 12, in <module>
    from mmlspark.lightgbm import LightGBMClassifier
  File "/usr/lib/anaconda2/envs/mmlspark/lib/python3.6/site-packages/mmlspark/lightgbm/LightGBMClassifier.py", line 11, in <module>
    from mmlspark.lightgbm._LightGBMClassifier import _LightGBMClassifier
ModuleNotFoundError: No module named 'mmlspark.lightgbm._LightGBMClassifier'

```

**python code**

``` python

# coding=UTF-8

import numpy as np
import pyspark

spark = pyspark.sql.SparkSession.builder.appName("spark lightgbm") \
    .config("spark.jars.packages", "com.microsoft.ml.spark:mmlspark_2.11:0.18.1") \
    .config("spark.jars.repositories", "https://mmlspark.azureedge.net/maven") \
    .getOrCreate()

import mmlspark
from mmlspark.lightgbm import LightGBMClassifier
from pyspark.ml.feature import VectorAssembler
# from pyspark.ml.tuning import CrossValidator, ParamGridBuilder
from pyspark.ml.evaluation import BinaryClassificationEvaluator
from pyspark.ml import Pipeline

df_train = spark.read.format("csv") \
  .option("inferSchema", "true") \
  .option("header", "true") \
  .option("sep", ",") \
  .load("/metrics_bak/mmlspark2.csv")

df_val = df_train

df_test = df_train

feature_cols = list(df_train.columns)
feature_cols.remove("label")  
assembler = VectorAssembler(inputCols=feature_cols, outputCol="features") 

lgb = LightGBMClassifier(
    objective="binary",
    boostingType='gbdt',
    isUnbalance=True,
    featuresCol='features',
    labelCol='label',
    maxBin=60,
    baggingFreq=1,
    baggingSeed=696,
    earlyStoppingRound=30,
    learningRate=0.1,
    lambdaL1=1.0,
    lambdaL2=45.0,
    maxDepth=3,
    numLeaves=128,
    baggingFraction=0.7,
    featureFraction=0.7,
    # minSumHessianInLeaf=1,
    numIterations=800,
    verbosity=30
)

stages = [assembler, lgb]

pipeline_model = Pipeline(stages=stages)
model = pipeline_model.fit(df_train)


train_preds = model.transform(df_train)
val_preds = model.transform(df_val)
test_preds = model.transform(df_test)


binaryEvaluator = BinaryClassificationEvaluator()
print ("Train AUC: " + str(binaryEvaluator.evaluate(train_preds, {binaryEvaluator.metricName: "areaUnderROC"})))
print ("Val AUC: " + str(binaryEvaluator.evaluate(val_preds, {binaryEvaluator.metricName: "areaUnderROC"})))
print ("Test AUC: " + str(binaryEvaluator.evaluate(test_preds, {binaryEvaluator.metricName: "areaUnderROC"})))


```

**Comment by rusonding** [comment]
--jars file:///root/.ivy2/jars/com.microsoft.ml.spark_mmlspark_2.11-1.0.0-rc1.jar,file:///root/.ivy2/jars/com.microsoft.ml.lightgbm_lightgbmlib-2.3.100.jar

**Comment by TomasZhang19** [comment]
Hello, could you please tell me how to get the "com.microsoft.ml.spark_mmlspark_2.11-1.0.0-rc1.jar"?

**Comment by ranadeepsingh** [comment]
This issue was caused by an incomplete installation of legacy mmlspark where auto-generated Python wrapper modules were missing. Modern published releases of synapseml (v1.0.0+ and v1.1.3) include all auto-generated Python base classes, resolving this missing module error. We are closing this issue as fixed in published releases.
